src/disasm/section.rs

```diff
@@ -0,0 +1,257 @@
+    fn add_instruction_multiple_symbols_multiple_instructions() {
+        let mut section = Section::new("sec");
+        section.add_symbol(Symbol::new("sym1"));
+        let _ = section.add_instruction(Instruction::new("nop", "", ""));
+        let _ = section.add_instruction(Instruction::new("mov", "-0x1198(%rbp),%rax", ""));
+        section.add_symbol(Symbol::new("sym2"));
+        let _ = section.add_instruction(Instruction::new("lea", "0x357d6(%rip),%rcx", ""));
+
+        let mut comparison_sym1 = Symbol::new("sym1");
+        comparison_sym1.add_instruction(Instruction::new("nop", "", ""));
+        comparison_sym1.add_instruction(Instruction::new("mov", "-0x1198(%rbp),%rax", ""));
+        let mut comparison_sym2 = Symbol::new("sym2");
+        comparison_sym2.add_instruction(Instruction::new("lea", "0x357d6(%rip),%rcx", ""));
+        assert_eq!(
+            section,
+            Section {
+                name: "sec".to_string(),
+                symbols: Vec::from([comparison_sym1, comparison_sym2]),
+            }
+        );
+    }
+
+    #[test]
+    fn sort_symbols() {
+        let mut section = Section::new("sec");
+        section.add_symbol(Symbol::new("sym2"));
+        section.add_symbol(Symbol::new("sym2"));
+        section.add_symbol(Symbol::new("sym1"));
+        section.add_symbol(Symbol::new("zsym"));
+        section.add_symbol(Symbol::new("asym"));
+        section.sort_symbols();
+        assert_eq!(
+            section,
+            Section {
+                name: "sec".to_string(),
+                symbols: Vec::from([
+                    Symbol::new("asym"),
+                    Symbol::new("sym1"),
+                    Symbol::new("sym2"),
+                    Symbol::new("sym2"),
+                    Symbol::new("zsym")
+                ]),
+            }
+        );
+    }
+
+    #[test]
+    fn to_string_unnamed_empty_section() {
+        let section = Section::new("");
+        assert_eq!(section.to_string(), ":\n".to_string())
+    }
+
+    #[test]
+    fn to_string_named_empty_section() {
+        let section = Section::new("sec");
+        assert_eq!(section.to_string(), "sec:\n".to_string())
+    }
+
+    #[test]
+    fn to_string_named_and_non_empty_section() {
+        let mut section = Section::new("sec");
+        section.add_symbol(Symbol::new("sym1"));
+        let add_instr_res = section.add_instruction(Instruction::new("nop", "", ""));
+        assert_eq!(add_instr_res, Ok(()));
+        let add_instr_res =
+            section.add_instruction(Instruction::new("mov", "-0x1198(%rbp),%rax", ""));
+        assert_eq!(add_instr_res, Ok(()));
+        section.add_symbol(Symbol::new("sym2"));
+        let add_instr_res =
+            section.add_instruction(Instruction::new("lea", "0x357d6(%rip),%rcx", ""));
+        assert_eq!(add_instr_res, Ok(()));
+
+        assert_eq!(
+            section.to_string(),
+            "\
+sec:
+    sym1:
+        nop
+        mov
+    sym2:
+        lea
+"
```

Comment:
Seems like a bit of an overkill since this is an exercise, however if you feel like it [this library](https://github.com/dtolnay/indoc) provides a macro for better multi-line string indentation handling

Reply:
I will look into it, thanks